From 93c55395fabd99dcbf4ec9697898464101214672 Mon Sep 17 00:00:00 2001
From: Ribesg <ribesg@yahoo.fr>
Date: Wed, 9 Apr 2014 15:02:31 +0200
Subject: [PATCH] Reduce CPU load of NBT compression

Based on a comment on https://bugs.mojang.com/browse/MC-33134 by Poweruser (2014-01-15/9:18PM).

Modified files:
- src/main/java/net/minecraft/server/NBTCompressedStreamTools.java

diff --git a/src/main/java/net/minecraft/server/NBTCompressedStreamTools.java b/src/main/java/net/minecraft/server/NBTCompressedStreamTools.java
index d22a480..669a13a 100644
--- a/src/main/java/net/minecraft/server/NBTCompressedStreamTools.java
+++ b/src/main/java/net/minecraft/server/NBTCompressedStreamTools.java
@@ -37,9 +37,15 @@ public class NBTCompressedStreamTools {
         try {
         DataOutputStream dataoutputstream = new DataOutputStream(new GZIPOutputStream(outputstream));
 
+        // Nukkit start
+        ByteArrayOutputStream byteArrayStream = new ByteArrayOutputStream();
+        DataOutputStream dataStream = new DataOutputStream(byteArrayStream);
         try {
-            a(nbttagcompound, (DataOutput) dataoutputstream);
+            a(nbttagcompound, (DataOutput) dataStream);
+            dataoutputstream.write(byteArrayStream.toByteArray());
         } finally {
+            dataStream.close();
+            // Nukkit end
             dataoutputstream.close();
         }
         } catch (IOException e) {
-- 
